packages/services/api/src/modules/audit-logs/providers/audit-log-recorder.ts

```diff
@@ -0,0 +1,77 @@
+import { randomUUID } from 'node:crypto';
+import { Injectable, Scope } from 'graphql-modules';
+import { captureException } from '@sentry/node';
+import { type User } from '../../../shared/entities';
+import { sql as c_sql, ClickHouse } from '../../operations/providers/clickhouse-client';
+import { Logger } from '../../shared/providers/logger';
+import { auditLogSchema, type AuditLogSchemaEvent } from './audit-logs-types';
+
+type AuditLogRecordEvent = AuditLogSchemaEvent & {
+  metadata: {
+    userId: string;
+    userEmail: string;
+    organizationId: string;
+    user: User;
+  };
+};
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+/**
+ * Responsible for recording audit log events and storing them in ClickHouse
+ */
+export class AuditLogRecorder {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogRecorder' });
+  }
+
+  async record(data: AuditLogRecordEvent): Promise<void> {
+    try {
+      const { eventType } = data;
+      const { organizationId, userEmail, userId } = data.metadata;
+      this.logger.debug('Creating a log audit event (data=%o)', data);
+
+      const parsedEvent = auditLogSchema.parse(data);
+      const metadata = {
+        user: data.metadata.user,
+        ...parsedEvent,
+      };
+
+      const eventMetadata = JSON.stringify(metadata);
+      const eventTime = new Date();
+      const id = randomUUID();
+      const values = [id, eventTime, userId, userEmail, organizationId, eventType, eventMetadata];
+
+      await this.clickHouse.insert({
+        query: c_sql`
+              INSERT INTO "audit_log" (
+                "id"
+                , "event_time"
+                , "user_id"
+                , "user_email"
+                , "organization_id"
+                , "event_action"
+                , "metadata"
+              )
+              FORMAT CSV`,
```

Comment:
Because we insert only one row at a time, we don't have to use `CSV` format. We could, but it has no performance value, it brings a bit of complexity (you need to wrap strings with `"`, booleans as `"1"` or `"0"` and nulls as `\\N`).
Let's insert data in an easier way: `INSER INTO audit_logs (...) VALUES (....)`.